app/src/main/java/com/amaze/filemanager/asynchronous/handlers/FileHandler.kt

```diff
@@ -101,6 +101,9 @@ class FileHandler(
             // there was no list view, means the directory was empty
             main.loadlist(main.currentPath, true, mainFragmentViewModel.openMode)
         }
+        main.currentPath?.let {
+            main.mainFragmentViewModel?.evictPathFromListCache(it)
```

Comment:
I don't know if it is possible, but it seems this should wait until the fragment is not null and then try to evict the path, maybe something like event based?

Reply:
Doesn't really matter. If fragment is null then the underlying viewmodel wouldn't be present as well. That automatically means there's no cache by the time fragment is available.